pallets/subtensor/src/weights.rs

```diff
@@ -40,11 +54,36 @@ impl<T: Config> Pallet<T> {
             Error::<T>::WeightsCommitNotAllowed
         );
 
-        WeightCommits::<T>::insert(
-            netuid,
-            &who,
-            (commit_hash, Self::get_current_block_as_u64()),
-        );
+        // Check if the nonce is greater than the last processed nonce
+        let last_processed_nonce = Self::get_last_processed_nonce(netuid, &who);
+        ensure!(nonce > last_processed_nonce, Error::<T>::NonMonotonicNonce);
```

Comment:
Could you brick your subnet by using a nonce that's u64 max? Should we enforce cur nonce + 1 instead?

Reply:
You couldnt brick your subnet by doing this , what would happen is that you would brick your validator , as the nonces are per validator, not subnet